libparsec/crates/client/tests/unit/workspace_inbound_sync_monitor.rs

```diff
@@ -0,0 +1,432 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+use std::collections::VecDeque;
+use std::sync::{Arc, Mutex};
+
+use libparsec_platform_async::prelude::*;
+use libparsec_tests_fixtures::prelude::*;
+use libparsec_types::prelude::*;
+
+use super::{
+    inbound_sync_monitor_loop, InboundSyncManagerIO, IncomingEvent, WaitForNextIncomingEventOutcome,
+};
+use crate::event_bus::AnySpiedEvent;
+use crate::workspace::{
+    InboundSyncOutcome, WorkspaceGetNeedInboundSyncEntriesError, WorkspaceSyncError,
+};
+
+/*
+ * Important stuff
+ */
+
+/// Optional callback to modify the outcome of the event
+type MaybeSideEffect<R> = Option<Box<dyn FnOnce(&mut R) + Send + Sync + 'static>>;
+/// Optional callback (taking an additional `expect_xxx` param) to modify the outcome of the event
+type MaybeSideEffectP1<P, R> = Option<Box<dyn FnOnce(P, &mut R) + Send + Sync + 'static>>;
+
+#[allow(dead_code)]
+enum InboundSyncMonitorEvent {
+    WaitForNextIncomingEvent {
+        outcome: WaitForNextIncomingEventOutcome,
+        #[allow(clippy::type_complexity)]
```

Comment:
Given, you have added the type alias for `MaybeSideEffect...`, you could remove the `allow(clippy::type_complexity)`